### PR TITLE
update dot to use srepr

### LINKED ISSUE #11634: dotprint doesn't use the correct order for x**2
The dot diagram in the tutorial is wrong (http://docs.sympy.org/dev/tutorial/manipulation.html). It shows 

```
          Pow
          /  \
Integer(2)    Symbol('x')
```

but it should show

```
           Pow
           /  \
Symbol('x')    Integer(2)
```

since it represents `x**2`, not `2**x`. 

I can't figure out how to make dot give the vertices in the right order. Whatever the fix is, we should fix this in the dot printer as well as the tutorial.

### PR DESCRIPTION
**Pull request by smichr**
#### References to other Issues or PRs

closes #10314
closes #11634

#### Brief description of what is fixed or changed

srepr is used in building the expression and does so in a non-`O(n**2)` fashion. It also respect commutativity in Mul.

#### Release Notes

- printing
    - dot should be faster at generating output and now respects commutativity

### CODE REVIEW

**Comment by smichr** [comment]
```python
>>> print(dotprint(A*B))
digraph{

# Graph style
"ordering"="out"
"rankdir"="TD"

#########
# Nodes #
#########

"Mul(Symbol('A', commutative=False), Symbol('B', commutative=False))_()" ["color
"="black", "label"="Mul", "shape"="ellipse"];
"Symbol('A', commutative=False)_(0,)" ["color"="black", "label"="A", "shape"="el
lipse"];
"Symbol('B', commutative=False)_(1,)" ["color"="black", "label"="B", "shape"="el
lipse"];

#########
# Edges #
#########

"Mul(Symbol('A', commutative=False), Symbol('B', commutative=False))_()" -> "Sym
bol('A', commutative=False)_(0,)";
"Mul(Symbol('A', commutative=False), Symbol('B', commutative=False))_()" -> "Sym
bol('B', commutative=False)_(1,)";
}
>>> print(dotprint(B*A))
digraph{

# Graph style
"ordering"="out"
"rankdir"="TD"

#########
# Nodes #
#########

"Mul(Symbol('B', commutative=False), Symbol('A', commutative=False))_()" ["color
"="black", "label"="Mul", "shape"="ellipse"];
"Symbol('B', commutative=False)_(0,)" ["color"="black", "label"="B", "shape"="el
lipse"];
"Symbol('A', commutative=False)_(1,)" ["color"="black", "label"="A", "shape"="el
lipse"];

#########
# Edges #
#########

"Mul(Symbol('B', commutative=False), Symbol('A', commutative=False))_()" -> "Sym
bol('B', commutative=False)_(0,)";
"Mul(Symbol('B', commutative=False), Symbol('A', commutative=False))_()" -> "Sym
bol('A', commutative=False)_(1,)";
}
```

I checked that the `2**x` and `x**2` print differently, too.

**Comment by asmeurer** [comment]
If the `x**2`/`2**x` issue is fixed we should revert the example in the tutorial back to `x**2`.

File: sympy/printing/dot.py [hunk 1]
```diff
@@ -7,27 +7,32 @@
...
 from sympy.core.mul import Mul
+from sympy.printing.repr import srepr
 
 __all__ = ['dotprint']
 
 default_styles = ((Basic, {'color': 'blue', 'shape': 'ellipse'}),
           (Expr,  {'color': 'black'}))
 
 
-sort_classes = (Add, Mul)
```
Comment (asmeurer) [inline sympy/printing/dot.py]: Why was `x**2` sorted then? Maybe a test should be added for it.

Comment (smichr) [reply sympy/printing/dot.py]: Here is `dotprint(2**x+x**3)`:

image

**Comment by smichr** [comment]
Thanks, @sylee957 .

**Pull request merged**